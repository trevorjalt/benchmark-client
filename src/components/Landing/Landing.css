.Landing__image {
    width: 60vw;
    border-radius: 20px;
    margin: 3vh 0 1vh 0;
    border: .4vw solid;
    border-color: #F78F38;
}

.Landing__image-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.LandingPageTitle {
    display: flex;
    justify-content: center;
}

.LandingPage__section-titles {
    color: #D11053;
    font-size: 3vh;
}

.Benchmark__description {
    background-color: #FEEFDD;
    width: 75vw;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 20px;
    margin: 4vh 0 4vh 0;
    padding: 3vh;
    font-size: 2.2vh;
    text-align: left;
    line-height: 3.2vh;
}

.Benchmark__functionality {
    color: #D11053;
    font-size: 2.5vh;
}

.Benchmark__title {
    font-size: 5.5vh;
    color: #D11053;
}

.Benchmark__tagline {
    font-size: 3.5vh;
    margin: 0 5vw 8vh 5vw;
}

.Description__container {
    display: flex;
    justify-content: center;

}

@media only screen and (min-width: 800px) {
    .Landing__image {
        width: 30vw;
        border-radius: 20px;
        margin: 3vh 0 1vh 0;
        border: .4vw solid;
        border-color: #F78F38;
    }
}
