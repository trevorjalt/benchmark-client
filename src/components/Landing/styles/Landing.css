img {
    height: 45vh;
}