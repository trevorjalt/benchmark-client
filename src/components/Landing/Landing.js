import React, { Component } from 'react'
import goku from './styles/images/goku.png'
import goku2 from './styles/images/goku2.png'
import goku3 from './styles/images/goku3.png'
import './styles/Landing.css'

export default class Landing extends Component {
    render() {
        return (
            <div>
                <h1>benchmark</h1>
                <h2>A sensible way to track your strength-training progress.</h2>
                <section>
                    <h3>Ditch your pen and paper</h3>
                    <img src={goku} alt='goku' />
                    <p>Here at benchmark we are strength-training enthusiasts.  We understand the amount of <em>stuff</em> you carry at the gym as you work to break your limits.  So we're here to help.  benchMark is an easy solution allowing you to leave that pen and paper at home, and quickly log and track your workout progress in the same device that plays those tunes to keep you pumped up.</p>
                </section>
                <section>
                    <h3>Keep your focus where it belongs</h3>
                    <p>We're all about no muss, no fuss.  benchMark keeps it simple, making it easy for you to:</p>
                    <img src={goku2} alt='goku' />
                    <p>Create custom workouts</p>
                    <img src={goku3} alt='goku' />
                    <p>Quickly analyze your progress.  You input numbers, you receive numbers.  No graphs to interpret.  Sensible.</p>
                </section>
            </div>
        )
    }
}