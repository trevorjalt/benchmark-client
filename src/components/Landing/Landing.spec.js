import React from 'react'
import Landing from './Landing'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`Landing Component`, () => {
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <Landing />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})