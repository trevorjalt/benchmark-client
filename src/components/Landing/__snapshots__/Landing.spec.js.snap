// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component renders without crashing 1`] = `
<div>
  <div
    className="LandingPageTitle"
  >
    <h1
      className="Benchmark__title"
    >
      benchMark
    </h1>
  </div>
  <h2
    className="Benchmark__tagline"
  >
    A sensible way to track your strength-training progress
  </h2>
  <section>
    <h3
      className="LandingPage__section-titles"
    >
      Ditch your pen and paper
    </h3>
    <img
      alt="notebook and pen"
      className="Landing__image"
      src="notebook-and-pen.png"
    />
    <div
      className="Description__container"
    >
      <div
        className="Benchmark__description"
      >
        <p>
          Here at benchmark we are strength-training enthusiasts.  We understand the amount of stuff you carry at the gym as you work to break your limits.  So we're here to help.  benchMark is an easy solution allowing you to leave that pen and paper at home, and quickly log and track your workout progress in the same device that plays those tunes to keep you pumped up.
        </p>
      </div>
    </div>
  </section>
  <section>
    <h3
      className="LandingPage__section-titles"
    >
      Keep your focus where it belongs
    </h3>
    <div
      className="Description__container"
    >
      <div
        className="Benchmark__description"
      >
        <p>
          We're all about no muss, no fuss.  benchMark keeps it simple, making it easy for you to:
        </p>
      </div>
    </div>
    <h4
      className="Benchmark__functionality"
    >
      Create custom workouts
    </h4>
    <div
      className="Landing__image-container"
    >
      <img
        alt="New workout screenshot"
        className="Landing__image"
        src="new-workout-screenshot.png"
      />
      <img
        alt="New workout input screenshot"
        className="Landing__image"
        src="new-workout-input-screenshot.png"
      />
    </div>
    <div
      className="Description__container"
    >
      <div
        className="Benchmark__description"
      >
        <p>
          benchMark gives you control over your experience.  Add exercises.  Add sets.  Complete them as you go, or complete them all at once.  Make a mistake?  No worries, you can update as you go.  benchMark is designed to let you focus on the task at hand: breaking your limits.
        </p>
      </div>
    </div>
    <h4
      className="Benchmark__functionality"
    >
      Quickly review your workout history
    </h4>
    <img
      alt="myWorkouts collapsed view screenshot"
      className="Landing__image"
      src="myworkouts-collapsed-view-screenshot.png"
    />
    <div
      className="Description__container"
    >
      <div
        className="Benchmark__description"
      >
        <p>
          benchMark displays a list of all your completed workouts.  Want a more detailed view?  Give it a double-tap to see your benchMark in its full glory.
        </p>
      </div>
    </div>
    <img
      alt="myWorkouts expanded view screenshot"
      className="Landing__image"
      src="myworkouts-expanded-view-screenshot.png"
    />
    <h4
      className="Benchmark__functionality"
    >
      Pick up where you left off
    </h4>
    <img
      alt="myWorkouts continue button screenshot"
      className="Landing__image"
      src="continue-button-screenshot.png"
    />
    <div
      className="Description__container"
    >
      <div
        className="Benchmark__description"
      >
        <p>
          Accidently close the app in the middle of a workout?  We got you.  Navigate over to the myWorkouts page, give your most recent workout a double-tap, and click on that 'Continue' button.  If you submit your set information as you complete it, it'll be saved, making it easy for you to quickly pick up right where you left off.
        </p>
      </div>
    </div>
  </section>
</div>
`;
